juju/juju
# Pruner workers

Requires https://github.com/juju/juju/pull/17343 to land first.

---

The action and status history pruners were originally using the
model config API to get the information. We don't want to do that
anymore. Instead, we want to use the service factory directly. This
allows for a more efficient streamlined juju. We can in theory drop
API methods as we don't need to expose all of the internal calls
via the API server.

The work is simple, use the new service factory from the model worker
changes (see https://github.com/juju/juju/pull/17343) and start to use the factory directly.

Note: this isn't moving over the actual pruning. That will need to be done later
when implementing actions (status history is going away so we don't need the
worker).

## Checklist




## QA steps

```
$ juju bootstrap lxd test
$ juju add-model default
$ juju model-config max-status-history-age=1m
```

Should see info message about the pruner config changing.

## Links

**Jira card:** JUJU-
